### DISCUSSION TITLE
Is the file availability guaranteed?

### DISCUSSION THREAD

**Discussion by stephenctw** [filecoin-project/community#337] [General]
Hi I am pretty new to this project. I am wondering if it's possible for the miner to keep proving to the blockchain that he has kept the file on storage, however refuse to transfer the file when someone else try to retrieve the file if there's greater interest to hide the file. I've read a few paper explaining the proof system filecoin is using, but I'm sure I found an explanation. Any pointers to useful documentation are very appreciated!

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
no

**Reply by JDLH** [reply]
This is a good question, I think. 

> no

I think you are saying, "no, the miner may not refuse to transfer the file when someone else tries to retrieve it". OK, that's good to know. What section of what spec guarantees that the client who paid to store the data, or someone performing a retrieval deal on behalf of that client, will have access to the data? And for what period do they have access — the last X hours of the storage contract? For Y hours after the end of the storage contract? For the whole duration of the storage contract?

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
what i am saying is: 

no, the file availability is not guaranteed. not for the one who stored the file, not for anybody else.

a storage provider decides whom to deliver data to. there is no part in the specs or in current "on chain" contracts that is related to retrieval. a storage provider can refuse to deliver data when ever he wants too.

**Reply by JDLH** [reply]
Thank you for the clarification. I had indeed misunderstood your answer.

> no, the file availability is not guaranteed. not for the one who stored the file, not for anybody else.

Then I don't understand part of the Filecoin economics. Part of why I might want to pay someone to store my data is because I want access to it in the future. If I make a storage deal with a storage miner to store my data for 180 days, how can I be sure that I will also be able to retrieve the data at the end of that time? In other words, how can I pay for *future availability* of my data, not just for its *storage*?

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
you can't for now. 

all you can do is choosing your storage providers carefully, store the data multiple times (the docs recommend 10 times) etc. in the end you will have to rely on legally binding agreements (aka paper contracts - or their digital equivalent) to get the service you desire.

**Reply by rjan90** [reply]
Until Filecoin gets smart contracts, and something similar to an SLA-agreement implemented with guarantees of retrievals, I would choose storage providers you trust, and know you can retrieve your files from 🙂

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
if that will ever happen, even with the fvm is questionable. we do a lot of hokuspokus to prove that a file is stored - we have little on proving that a file got transferred from miner to client
